src/main/java/com/zufar/onlinestore/product/endpoint/ProductsEndpoint.java

```diff
@@ -25,40 +28,38 @@
 @Validated
 @RequestMapping(value = "/api/products")
 public class ProductsEndpoint {
+
     private final ProductInfoRepository productInfoRepository;
     private final ProductInfoDtoConverter productInfoDtoConverter;
+    private final ProductApi productInfoService;
 
     @GetMapping("/{id}")
     public ResponseEntity<ProductInfoDto> getProductInfoById(@PathVariable("id") final String id) {
         log.info("Received request to get the ProductInfo with id - {}.", id);
-        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(Integer.parseInt(id));
+        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(UUID.fromString(id));
         if (ProductInfo.isEmpty()) {
             log.info("the ProductInfo with id - {} is absent.", id);
             return ResponseEntity.notFound()
                     .build();
         }
-        ProductInfoDto ProductInfoDto = productInfoDtoConverter.convertToDto(ProductInfo.get());
+        ProductInfoDto ProductInfoDto = productInfoDtoConverter.toDto(ProductInfo.get());
         log.info("the ProductInfo with id - {} was retrieved - {}.", id, ProductInfoDto);
         return ResponseEntity.ok()
                 .body(ProductInfoDto);
     }
 
     @GetMapping
-    public ResponseEntity<Collection<ProductInfoDto>> getAllProducts() {
-        log.info("Received request to get all ProductInfos");
-        Collection<ProductInfo> productInfoCollection = productInfoRepository.findAll();
+    public ResponseEntity<Page<ProductResponseDto>> getAllProducts(Pageable pageable) {
```

Comment:
Why you return Page<ProductResponseDto> instead of Collection<ProductResponseDto> ?

Reply:
Pageable and Page is a data layer interfaces and should be used only in Service and Repository